### PR TITLE
Cesium 3DTilers driver.

### PR DESCRIPTION
**Pull request by danlipsa**
It needs:
- proj updated to version > 5
- VTK rebuilt from Cesium3DTilesWriter branch: https://gitlab.kitware.com/core3d/danesfield-conda-recipes/-/merge_requests/2

### CODE REVIEW

**Comment by danlipsa** [comment]
@jacobderosa @mleotta Please review.

**Comment by danlipsa** [comment]
See also
https://gitlab.kitware.com/core3d/danesfield-conda-recipes/-/merge_requests/2

**Comment by danlipsa** [comment]
@jacobderosa Please review.  Let me know when any of the requirements listed at the top are done (new VTK and proj)

**Comment by danlipsa** [comment]
@mleotta @jacobderosa I hard coded the UTM zone to Jacksonville: 17 N. We did not store the UTM zone in the OBJs so we'll need add that probably when we read images or point clouds.

File: tools/run_danesfield.py [hunk 1]
```diff
@@ -490,6 +490,24 @@ def main(args):
              'texture-mapping',
              cmd_args)
 
+    #############################################
+    # 3D Tiles Generation
+    #############################################
+    tiler_outdir = os.path.join(working_dir, 'tiler')
+    input_tiler = glob.glob(os.path.join(texture_mapping_outdir, "*.obj"))
+    utm_zone = 17        # jacksonville dataset
+    utm_hemisphere = "N"
```
Comment (dmjoy) [inline tools/run_danesfield.py:499]: I don't think we should have to hardcode anything here as this information should be coming through the configuration files (or we should be able to derive that info from what's in the config).  (Perhaps it's in the VisSat config @jacobderosa ?)

File: Dockerfile [hunk 2]
```diff
@@ -125,6 +126,13 @@ RUN ["/bin/bash", "-c", "source /opt/conda/etc/profile.d/conda.sh && \
   conda activate core3d && \
   pip install -e ./danesfield"]
 
+# Install 3d-tiles-tools for converting glb to b3dm
+RUN ["/bin/bash", "-c", "git clone https://github.com/CesiumGS/3d-tiles-validator.git"]
+
+# Install gltf-pipeline for converting gltf to glb
+RUN ["/bin/bash", "-c", "git clone https://github.com/CesiumGS/gltf-pipeline.git"]
```
Comment (dmjoy) [inline Dockerfile:133]: Is it necessary / desirable to install these (beyond just cloning) here also?  I see installation is covered in the `tiler-readme.md` document.  Or is installation largely only necessary in certain niche use cases.

**Review by dmjoy — COMMENTED** [review]
This largely LGTM from an integration standpoint.  I think @jacobderosa is going to take a look as well.

File: tools/run_danesfield.py (hunk 1 above)
Comment (danlipsa) [reply tools/run_danesfield.py:499]: I agree. However this is what we have so far afaik. We'll need code to get the the zone from images (or point clouds) and propagate it here.

Comment (jacobderosa) [reply tools/run_danesfield.py:499]: UTM information is included in the VisSat config file. Though that won't be helpful if a user already has a point cloud and doesn't need to run the VisSat step

File: Dockerfile (hunk 2 above)
Comment (danlipsa) [reply Dockerfile:133]: Good catch. Not sure why I omitted that - I'll check if this is required and add it here or update the documentation.

File: tools/run_danesfield.py (hunk 1 above)
Comment (danlipsa) [reply tools/run_danesfield.py:499]: Full georeference information should be read from the image or point cloud files and then propagated to the mesh (building model). I think even a vissat config file is a hack. What happens if you process an image from a different UTM zone?

File: Dockerfile (hunk 2 above)
Comment (danlipsa) [reply Dockerfile:133]: Yes, npm install is needed - it downloads all dependencies.

File: tools/run_danesfield.py (hunk 1 above)
Comment (danlipsa) [reply tools/run_danesfield.py:499]: @mleotta We should discuss this in our meeting to decide what we do.

**Comment by danlipsa** [comment]
@dmjoy @jacobderosa Ready to merge?

**Comment by danlipsa** [comment]
I added the code for running 3dtiles in the imageless pipeline. Ready to merge?

**Comment by dmjoy** [comment]
It all looks good to me apart from the hardcoded zone information.  I'm hesitant to merge into master as such.  I think in the near term (for the demo) we can just build a Docker image from this branch.  Beyond that I think we either need to derive that information or put it into the config / arguments.

**Comment by danlipsa** [comment]
Sounds good. I can see that for vissat we aready have that from the json file that points to the images. for p3d and for point clouds I can take a stab at this after the demo.

**Comment by danlipsa** [comment]
On a second thought, maybe a way to commit this would be to pass the UTM zone and hemisphere as command line. This parameter would be needed only if we don't pass vissat and it would be similar with what vissat does - provide the UTM zone and hemisphere as a separate parameter instead of extracting that from the image file. Thoughts?

**Comment by mleotta** [comment]
The UTM zone is already in the input (possibly indirectly, since the ROI may be expressed in Lat-long from which you can extract the UTM zone).  We just need to make sure it gets pass through everywhere it is needed.  There should be no need to manually specify the UTM zone.

**Comment by danlipsa** [comment]
Sounds good.

**Comment by danlipsa** [comment]
@mleotta My thinking was that we might need to pass the UTM zone anyway. For instance, the point cloud input (a PLY) might be just UTM values without the zone or hemisphere.

**Comment by mleotta** [comment]
I believe the LAS files we produce from either P3D or VisSat have complete geocoordinates.  The points may be stored as just two coordinates (easting/northing) in UTM, but the file itself has the zone somewhere in the header.

**Comment by danlipsa** [comment]
Sounds good.

**Comment by danlipsa** [comment]
@dmjoy @mleotta Please review the UTM zone and hemisphere computation from the DSM file.

**Comment by dmjoy** [comment]
UTM Zone / hemisphere computation additions LGTM

**Pull request merged**